doc/index.rst

```diff
@@ -6,23 +6,113 @@
+This package contains code to read environmental sensor data and documentation for those sensors.
 
-Contributing
-============
+Overview
+========
 
-``lsst.ts.ess.common`` is developed at https://github.com/lsst-ts/ts_ess_common.
-You can find Jira issues for this module using `labels=ts_ess_common <https://jira.lsstcorp.org/issues/?jql=project%3DDM%20AND%20labels%3Dts_ess_common>`_.
+The ESS (Environmental Sensors Suite) reports environmental data, such as temperature, pressure, rain, lightning, and even acceleration.
+Note that the ESS is not the only reporter of environmental data; some systems such as cameras and the TMA have their own sensors and report that data directly.
+But where practical, we use the ESS, in order to provide a unified interface.
+
+The ESS CSC is coded in `ts_ess_csc`_.
+It is an indexed CSC, and each instance reads a specific set of sensors.
+
+Terminology
+-----------
+
+* Sensor: a device that measures one or more environmental parameters, such as temperature, pressure, or vibration.
+  One "sensor" includes all the different channels of data reported by the device that serves the data.
+  For example, multi-channel thermal sensors have multiple temperature probes, and the Omega HX85BA sensor measures relative humidity, air temperature, and barometric pressure.
+
+* Data client: software (subclasses of `data_client.BaseDataClient`) that connects to a data server, reads the data, and reports it using SAL.
+  Each ESS CSC instance is a collection of data clients that are specified and configured by the CSC's configuration.
+
+Sensors
+=======
+
+Most sensors serve their own data (directly or by being plugged into an electronics box provided by the vendor).
+Protocols include RS-232, RS-485, TCP/IP, USB, and FTDI.
+
+* Sensors that serve their data via TCP/IP are be directly read by an ESS instance running on Kubernetes.
+  These sensors are read by a subclass of `data_client.BaseDataClient`.
+
+* Sensors that serve their data via USB or FTDI cannot be directly read by an ESS instance, because software running on Kubernetes cannot read these protocols.
+  In this case we run custom software (in `ts_ess_controller`_) on RPis to read the data and transmit it via TCP/IP.
+  For this one case, we represent sensors using a subclass of `sensor.BaseSensor` and read data using a subclass of `device.BaseDevice`.
+  
+* Sensors that serve their data via RS-232 or RS-485 serial are served in two different ways:
+
+  * Some sensors are read by a Raspberry Pi running custom software, as per the "USB or FTDI" case above.
+    This includes sensors that receive power from a custom serial cable: our custom Raspberry Pis systems provide power on their RS-232 ports.
+
+  * Other sensors are read by serial-to-ethernet adapters, as per the TCP/IP case mentioned above.
+    This is our preferred technique for sensors that are powered in the conventional way.
```

Comment:
Again, not sure if this is the `preferred` way. That depends on the use case.

Reply:
I think it is preferred for new *kinds* of sensors, as it avoids the need for the relay server.

This will change if we run ESS instances inside RPis, but this is the documentation for the system we have now.

Reply:
I don't disagree that that probably would be preferred. But I don't think that that may necessarily apply to the sensors used on the inside of the MTDome, for example. Brian seems to want to use RPis for those. Once again, I think we should bring this up in the EAS meeting next week.

Reply:
OK. I'll remove that one line.